heuristic/classes/Stack.py

```diff
@@ -3,6 +3,7 @@
 from collections import deque
 from typing import Deque, List, Set
 
+from ..constants import DEPOT
```

Comment:
`heuristic.constants` 

I prefer fixed hierarchies rather than relative imports once they go above the current directory - so `from .Item import Item` is fine, but `from ..constants import DEPOT` has too many indirections, and `from heuristic.constants import DEPOT` is easier to read.